backend/src/layer/data/database/request.rs

```diff
@@ -0,0 +1,451 @@
+            Line,
+            "SELECT line_id as id, name, canteen_id FROM line WHERE line_id = $1",
+            id
+        )
+        .fetch_optional(&self.pool)
+        .await
+        .map_err(Into::into)
+    }
+
+    async fn get_lines(&self, canteen_id: Uuid) -> Result<Vec<Line>> {
+        sqlx::query_as!(
+            Line,
+            "SELECT line_id as id, name, canteen_id FROM line WHERE canteen_id = $1 ORDER BY position",
+            canteen_id
+        )
+        .fetch_all(&self.pool)
+        .await
+        .map_err(Into::into)
+    }
+
+    async fn get_meal(&self, id: Uuid, line_id: Uuid, date: Date) -> Result<Option<Meal>> {
+        sqlx::query!(
+            r#"
+            SELECT food_id, name, food_type as "meal_type: MealType",
+                price_student, price_employee, price_guest, price_pupil, serve_date as date, line_id,
+                new, frequency, last_served, next_served, average_rating, rating_count
+            FROM meal_detail JOIN food_plan USING (food_id)
+            WHERE food_id = $1 AND line_id = $2 AND serve_date = $3
+            "#,
+            id,
+            line_id,
+            date
+        )
+        .fetch_optional(&self.pool)
+        .await?
+        .map(|m| {
+            Ok(Meal {
+                id: null_error!(m.food_id),
+                line_id: m.line_id,
+                date: m.date,
+                name: null_error!(m.name),
+                meal_type: null_error!(m.meal_type),
+                price: Price {
+                    price_student: m.price_student as u32,
+                    price_employee: m.price_employee as u32,
+                    price_guest: m.price_guest as u32,
+                    price_pupil: m.price_pupil as u32
+                },
+                frequency: null_error!(m.frequency) as u32,
+                new: null_error!(m.new),
+                last_served: m.last_served,
+                next_served: m.next_served,
+                average_rating: null_error!(m.average_rating),
+                rating_count: null_error!(m.rating_count) as u32,
+            })
+        }).transpose()
+    }
+
+    async fn get_meals(&self, line_id: Uuid, date: Date) -> Result<Option<Vec<Meal>>> {
+        // If date too far into the future, return `None`.
+        // This should probably be inside the logic layer which currently does not exists for request.
+        let today = Local::now().date_naive();
+        let age = today - date;
+        if age.num_weeks() > MAX_WEEKS_DATA {
+            return Ok(None);
+        }
+
+        let meal = sqlx::query!(
+            r#"
+            SELECT food_id, name, food_type as "meal_type: MealType",
+                price_student, price_employee, price_guest, price_pupil, serve_date as date, line_id,
+                new, frequency, last_served, next_served, average_rating, rating_count
+            FROM meal_detail JOIN food_plan USING (food_id)
+            WHERE line_id = $1 AND serve_date = $2
```

Comment:
Consider adding some ordering here. This makes the tests easier and the frontend more consistent